### PR TITLE
fix[pubsub]: Remove the existing key rollover callback before creatin…

### PR DESCRIPTION
**Pull request by maciek5991**
…g a new one

### CODE REVIEW

**Comment by jpfr** [comment]
Thanks.
This could be handled inside `UA_PubSubKeyStorage_addKeyRolloverCallback`.
Then other users of this method have the same protection...

File: src/pubsub/ua_pubsub_keystorage.c [hunk 1]
```diff
@@ -165,6 +165,12 @@ UA_PubSubKeyStorage_addKeyRolloverCallback(UA_PubSubManager *psm,
     if(!psm || !ks || !callback || timeToNextMs <= 0)
         return UA_STATUSCODE_BADINVALIDARGUMENT;
 
+    // First remove the existing callback
+    if(*callbackID != 0) {
```
Comment (jpfr) [inline src/pubsub/ua_pubsub_keystorage.c:169]: On the contrary. When the callback already exists, then we don't have to re-add it and can do an early exit.

**Review by jpfr — COMMENTED** [review]
Comment inline.
Please rebase this for 1.4 and we pull it up to master from there.

File: src/pubsub/ua_pubsub_keystorage.c (hunk 1 above)
Comment (maciek5991) [reply src/pubsub/ua_pubsub_keystorage.c:212]: I don't agree. By not re-registering the callback, we risk missing timing adjustment.

Comment (maciek5991) [reply src/pubsub/ua_pubsub_keystorage.c:212]: ok, I agree. We don't have to re-add it

**Comment by jpfr** [comment]
👍

**Pull request merged**